pkg/library/projects/clone.go

```diff
@@ -41,12 +42,54 @@ type Options struct {
 	Env        []corev1.EnvVar
 }
 
+func ValidateAllProjects(workspace *dw.DevWorkspaceTemplateSpec) error {
+	// Temporary additional validation: a project cannot share a name with a dependentProject, and clonePaths cannot collide
+	// Map of project names to project sources (project, dependentProject, starterProject)
+	projectNames := map[string][]string{}
+	// Map of project clone paths to project sources ("project <name>", "starterProject <name>", "dependentProject <name>")
+	clonePaths := map[string][]string{}
+
+	for idx, project := range workspace.Projects {
+		projectNames[project.Name] = append(projectNames[project.Name], "projects")
+		clonePath := GetClonePath(&workspace.Projects[idx])
+		clonePaths[clonePath] = append(clonePaths[clonePath], fmt.Sprintf("project %s", project.Name))
+	}
+
+	for idx, project := range workspace.DependentProjects {
+		projectNames[project.Name] = append(projectNames[project.Name], "dependentProjects")
+		clonePath := GetClonePath(&workspace.Projects[idx])
+		clonePaths[clonePath] = append(clonePaths[clonePath], fmt.Sprintf("dependentProject %s", project.Name))
+	}
+
+	starterProject, err := GetStarterProject(workspace)
+	if err != nil {
+		return err
+	}
+	if starterProject != nil {
+		projectNames[starterProject.Name] = append(projectNames[starterProject.Name], "starterProjects")
+		clonePaths[starterProject.Name] = append(clonePaths[starterProject.Name], fmt.Sprintf("starterProject %s", starterProject.Name))
```

Comment:
Not necessary, but might be worth adding a comment (that's already been copy-pasted around) about starterProjects not having a clonePath field, so the name is used instead. In case someone mistakes this for a bug.

`// Starter projects do not allow specifying a clone path, so the name is used instead.`

Reply:
Added a comment here.